core/ts/Poll.ts

```diff
@@ -600,44 +647,60 @@ export class Poll implements IPoll {
   private genProcessMessagesCircuitInputsPartial = (index: number): CircuitInputs => {
     const messageBatchSize = this.batchSizes.messageBatchSize;
 
-    assert(index <= this.messages.length);
-    assert(index % messageBatchSize === 0);
+    assert(index <= this.messages.length, "The index must be <= the number of messages");
+    assert(index % messageBatchSize === 0, "The index must be a multiple of the message batch size");
 
+    // fill the msgs array with a copy of the messages we have
+    // plus empty messages to fill the batch
     let msgs = this.messages.map((x) => x.asCircuitInputs());
     while (msgs.length % messageBatchSize > 0) {
       msgs.push(msgs[msgs.length - 1]);
     }
-
+    // we only take the messages we need for this batch
     msgs = msgs.slice(index, index + messageBatchSize);
 
+    // fill the commands array with a copy of the commands we have
     let commands = this.commands.map((x) => x.copy());
     while (commands.length % messageBatchSize > 0) {
       commands.push(commands[commands.length - 1]);
     }
+    // we only take the commands we need for this batch
     commands = commands.slice(index, index + messageBatchSize);
 
     while (this.messageTree.nextIndex < index + messageBatchSize) {
       this.messageTree.insert(this.messageTree.zeroValue);
     }
 
+    // generate the path to the subroot of the message tree for this batch
     const messageSubrootPath = this.messageTree.genMerkleSubrootPath(index, index + messageBatchSize);
 
-    assert(IncrementalQuinTree.verifyMerklePath(messageSubrootPath, this.messageTree.hashFunc) === true);
+    assert(
+      IncrementalQuinTree.verifyMerklePath(messageSubrootPath, this.messageTree.hashFunc) === true,
+      "The message subroot path is invalid",
+    );
 
+    // validate that the batch index is correct, if not fix it
+    // this means that the end will be the last message
     let batchEndIndex = index + messageBatchSize;
     if (batchEndIndex > this.messages.length) {
       batchEndIndex = this.messages.length;
     }
 
+    // copy the public keys, pad the array with the last keys if needed
     let encPubKeys = this.encPubKeys.map((x) => x.copy());
     while (encPubKeys.length % messageBatchSize > 0) {
       encPubKeys.push(encPubKeys[encPubKeys.length - 1]);
     }
+    // then take the ones part of this batch
```

Comment:
Not quite understand this description? take some of the `encPubKeys` part of this batch?

Reply:
yup so each message has its corresponding encPubKey - we take the messages and keys part of the batch we are currently processing